### DISCUSSION TITLE
Open sessions when opening a directory

### DISCUSSION THREAD

**Discussion by Dioswilson** [olimorris/persisted.nvim#93] [Ideas]
Hello, I was searching for a plugin like this for a while, it all works well except one thing. I'd like to have the hability to open a directory (something like `nvim .config/nvim`) and have the session corresponding to that directory, this would be great because it would save time when opening a project.
While I have it implemented with the following autocmd:
```lua
vim.api.nvim_create_autocmd("VimEnter", {
    callback = function()
        if vim.fn.isdirectory(vim.fn.expand("%:p")) == 1 then
            vim.cmd("SessionLoad")
        end
    end,
})
```
This doesn't work too well, It sometimes doesn't work and other times it opens on the prevois buffer I was on when I closed(I have autosave on) and the buffer I was in goes all white and with some commands not working there. Also this doesn't behve well with neo-tree's `hijack_netrw_behavior`

**Comment by neandrake** [comment]
@Dioswilson I created a PR for this. Would you mind pulling the changes to test out if it results in the expected behavior?

https://github.com/olimorris/persisted.nvim/pull/104

**Comment by Dioswilson** [comment]
@neandrake I had some issues with it:
When  Neotree's `hijack_netrw_behavior` is on 
- When opening with `nvim .` it creates a new(emtpy) buffer with the file explorer
- When opening with `nvim dir` it creates a new(emtpy) buffer with neotree's explorer at the side (As it was before your change)

When opening opening without Neotree's option then a `nvim` buffer with the filesystem explorer opens (not focused, but present)

If launching from another directory (`nvim .config/nvim` for example) sometimes it tries to open current dir, and after that the directory I was trying to open gets a new session (previous session lost)
